### PR TITLE
Failed subscription handling

### PR DESCRIPTION
**Pull request by joeywinsford**
Include a callback for failed subscription requests to allow consumers to retry subscribing

### CODE REVIEW

**Comment by joeywinsford** [comment]
To clarify, we've had issues when our ES cluster has gone down or restarted, and it will accept connections a few seconds before it is ready to accept subscriptions.

This callback allows us to catch when the server refuses a subscription retry at a later interval.

The .NET API includes logic for retrying in the client, however that's a larger change that should probably be addressed along with connection dropping and retrying, so we've kept the client naive in this change.

File: README.md [hunk 1]
```diff
@@ -55,6 +55,7 @@ Subscribes to a stream to receive notifications as soon as an event is written t
 * onConfirmed - A function to be called when the server has confirmed that the subscription is running (function, takes in an [ISubscriptionConfirmation]((#isubscriptionconfirmation-interface)))
 * onDropped - A function to be called when the subscription is cancelled (function, takes in an [ISubscriptionDropped](#isubscriptiondropped-interface))
 * credentials - The user name and password needed for permission to subscribe to the stream ([ICredentials](#icredentials-interface), optional)
+* onNotHandled - A function to be called when the request for subscribition is not handled (function, takes in an [ISubscriptionNotHandled](#isubscriptionnothandled-interface))
```
Comment (x-cubed) [inline README.md]: subscribition => subscription

**Comment by x-cubed** [comment]
Thanks for the contribution!

Just fix up the spelling error and it's good to go.

**Comment by x-cubed** [comment]
Cool, thanks Joey

**Pull request merged**